### PR TITLE
🛁 Use XC-20 standard for `contracts-watr` TrueUSD

### PR DESCRIPTION
**Pull request by abam-iksde**

### CODE REVIEW

**Review by akoszowski — COMMENTED** [review]
Added a few comments here, before closing: https://github.com/trusttoken/contracts-pre22/pull/1234

**Comment by tt-krzysztof** [comment]
`contracts/common/ERC20.sol` can be safely removed.

**Comment by tt-krzysztof** [comment]
For redemptions to work, we also need to call `setBurnBounds` on the token (or through controller). [Eth mainnet values](https://etherscan.io/token/0x0000000000085d4780B73119b644AE5ecd22b376#readProxyContract) seem to be 1k-1bn.

File: packages/contracts-watr/contracts/common/XC20Wrapper.sol [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        IERC20Plus(nativeToken).mint(recipient, amount);
+    }
+
+    function name() public pure virtual returns (string memory) {
+        return "";
+    }
+
+    function symbol() public pure virtual returns (string memory) {
+        return "";
+    }
```
Comment (tt-krzysztof) [inline packages/contracts-watr/contracts/common/XC20Wrapper.sol:107]: Perhaps those can be abstract / left unimplemented?

File: packages/contracts-watr/contracts/interface/IERC20Plus.sol
Comment (tt-krzysztof) [inline packages/contracts-watr/contracts/interface/IERC20Plus.sol:1]: Perhaps we could include a reference to https://github.com/AstarNetwork/astar-frame/blob/polkadot-v0.9.39/precompiles/assets-erc20/ERC20.sol

**Review by tt-krzysztof — APPROVED** [review]

**Pull request merged**